internal/devices/create.go

```diff
@@ -64,62 +66,127 @@ func (c *Client) Create() *cobra.Command {
   metal device create -p $METAL_PROJECT_ID -P c3.medium.x86 -m sv -H test-rocky -O rocky_8 -r 47161704-1715-4b45-8549-fb3f4b2c32c7`,
 
 		RunE: func(cmd *cobra.Command, args []string) error {
-			var endDt *packngo.Timestamp
-
 			if userdata != "" && userdataFile != "" {
 				return fmt.Errorf("either userdata or userdata-file should be set")
 			}
 			cmd.SilenceUsage = true
-
 			if userdataFile != "" {
 				userdataRaw, readErr := os.ReadFile(userdataFile)
 				if readErr != nil {
-					return fmt.Errorf("Could not read userdata-file: %w", readErr)
+					return fmt.Errorf("could not read userdata-file: %w", readErr)
 				}
 				userdata = string(userdataRaw)
 			}
+			var endDt time.Time
 
 			if terminationTime != "" {
 				parsedTime, err := time.Parse(time.RFC3339, terminationTime)
 				if err != nil {
-					return fmt.Errorf("Could not parse time %q: %w", terminationTime, err)
+					return fmt.Errorf("could not parse time %q: %w", terminationTime, err)
 				}
-				endDt = &packngo.Timestamp{Time: parsedTime}
+				endDt = parsedTime
+			}
+			var publicSubnet int32
+			// In case of no configuration,
+			if publicIPv4SubnetSize == 0 {
+				publicSubnet = 31 // Assigin default to 31
+			} else {
+				publicSubnet = int32(publicIPv4SubnetSize)
 			}
 
+			formatedCdata := make(map[string]interface{})
+			if customdata != "" {
+				err := json.Unmarshal([]byte(customdata), &formatedCdata)
```

Comment:
You could try passing in a JSON string to test this out, e.g. `--customdata '{"testDate":"fromCli"}'` and see what happens.  @displague might be able to point you to more useful/meaningful examples.  I think unmarshalling the string is the right approach, but if we run into issues or just want to stick with the old way of passing the string directly to the API, that will require a spec change.

Reply:
It's successfully create devices, and it deletes device once it came to running state.

Reply:
Customer data is working fine with packngo, Metal-go is failed

Reply:
## Packngo
```
Vasubabus-MacBook-Pro:cluster-api-provider-packet vasubabu$ metal device create -p 63e0f0b2-f269-4482-9f56-c1d6c4d87b3b -P c3.small.x86 -m da -O ubuntu_20_04 -H create-device-dev-cdata --customdata '{"testDate":"fromCli"}'
2023/08/17 22:46:02
=======[REQUEST]=============
POST /metal/v1/projects/63e0f0b2-f269-4482-9f56-c1d6c4d87b3b/devices HTTP/1.1
Host: api.equinix.com
User-Agent: metal-cli/v0.15.0 packngo/v0.29.0
Connection: close
Content-Length: 237
Accept: application/json
Content-Type: application/json
X-Auth-Token: **REDACTED**
X-Consumer-Token: Equinix Metal CLI
Accept-Encoding: gzip

{
 "hostname": "create-device-dev-cdata",
 "plan": "c3.small.x86",
 "metro": "da",
 "operating_system": "ubuntu_20_04",
 "billing_cycle": "hourly",
 "project_id": "******************",
 "tags": [],
 "customdata": "{\"testDate\":\"fromCli\"}"
}
```

## Metal-go
```
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal device create -p 63e0f0b2-f269-4482-9f56-c1d6c4d87b3b -P c3.small.x86 -m da -O ubuntu_20_04 -H create-device-cdata --customdata '{"testDate":"fromCli"}'
2023/08/17 22:47:27
POST /metal/v1/projects/63e0f0b2-f269-4482-9f56-c1d6c4d87b3b/devices HTTP/1.1
Host: api.equinix.com
User-Agent: metal-cli/21d704b metal-go/0.17.0
Content-Length: 301
Accept: application/json
Content-Type: application/json
X-Auth-Token: ****************************
Accept-Encoding: gzip

{"billing_cycle":"hourly","customdata":{"testDate":"fromCli"},"hostname":"create-device-cdata","metro":"da","operating_system":"ubuntu_20_04","plan":"c3.small.x86","public_ipv4_subnet_size":31,"spot_instance":false,"spot_price_max":0,"tags":[],"termination_time":"0001-01-01T00:00:00Z","userdata":""
```
 In Spec side both are map[string]interface{}

Reply:
What is the error message with `metal-go` when you provide customdata?  Only the requests are provided above; what are the APi responses?

Reply:
Yes, our requests is approved, Devices are provisioned and come to Active state, and devices are deleted suddenly

Reply:
Looping back here from Slack:  it appears the issue here is that there is [no check that a `termination_time` was provided before calling `SetTerminationTime`](https://github.com/equinix/metal-cli/pull/282/files#diff-6153c670828ec05635bb32b940eab661ce8225f32ca7d5152b9c4d1b8576e1b9R133), so the device is being deleted automatically shortly after provisioning because we provided a `termination_time` that is in the past.